### PR TITLE
Creating remove git item function

### PR DESCRIPTION
**Pull request by scott-pickthorn**
- Remove-GitItem function imitates same functionality as `git rm` by deleting files from both local workspace and staging the files to be deleted from the repository

### CODE REVIEW

File: LibGit2/Functions/Remove-GitItem.ps1 [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        [String[]]
+        # The paths to the files/directories to remove in the next commit.
+        $Path,
+
+        [string]
+        # The path to the repository where the files should be removed. The default is the current directory as returned by Get-Location.
+        $RepoRoot = (Get-Location).ProviderPath
+    )
+    
+    Set-StrictMode -Version 'Latest'
```
Comment (splatteredbits) [inline LibGit2/Functions/Remove-GitItem.ps1:44]: Add `Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState` after this line.

File: Tests/Remove-GitItem.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    ThenFileShouldBeStaged -Path 'foo.bar'
+    ThenFileShouldBeDeleted -Path 'foo.bar'
+    ThenNoErrorShouldBeThrown
+}
+
+Describe 'Remove-GitItem.When File doesnt exist in the repository' {
+    GivenRepositoryWithFile -Name 'a.file'
+    GivenFileToStage  -Name 'different.file'
+    WhenFileIsStaged
+    ThenFileShouldNotBeStaged -Path 'different.file'
```
Comment (splatteredbits) [inline Tests/Remove-GitItem.Tests.ps1:190]: Add `ThenNoErrorShouldBeThrown` assertion.

File: LibGit2/Functions/Remove-GitItem.ps1 [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    $repo = Find-GitRepository -Path $RepoRoot -Verify
+
+    if( -not ((Test-Path -Path 'variable:repo') -and $repo) )
```
Comment (splatteredbits) [inline LibGit2/Functions/Remove-GitItem.ps1]: Why `Test-Path`?

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: LibGit2/Functions/Remove-GitItem.ps1 (hunk 1 above)
Comment (scott-pickthorn) [reply LibGit2/Functions/Remove-GitItem.ps1:44]: Done

File: Tests/Remove-GitItem.Tests.ps1 (hunk 2 above)
Comment (scott-pickthorn) [reply Tests/Remove-GitItem.Tests.ps1:190]: Done

File: LibGit2/Functions/Remove-GitItem.ps1 (hunk 3 above)
Comment (scott-pickthorn) [reply LibGit2/Functions/Remove-GitItem.ps1]: Removed `Test-Path`

File: Tests/Remove-GitItem.Tests.ps1 (hunk 2 above)
Comment (splatteredbits) [reply Tests/Remove-GitItem.Tests.ps1:190]: @scott-pickthorn Looks like the build failed. Also, update the release notes.

Comment (scott-pickthorn) [reply Tests/Remove-GitItem.Tests.ps1:190]: Done, added release notes and build is fixed

**Review by splatteredbits — APPROVED** [review]

**Pull request merged**